### PR TITLE
🧱 Updated starbound egg install script

### LINKED ISSUE #1827: [Bug]: starbound not installing mods
### Panel Version

1.10.1

### Wings Version

1.7.0

### Service

starbound

### Modified

No, I did not modify the egg

### Expected Behavior

If I enable the `WORKSHOP` setting, it should install and mount every subscribed mod so that the server loads them

### Actual Behavior

Some mods do not have a contents.pak file. Instead, the .pak file is named diferently. In one case, we observed the file name was the name of the mod. In another case, it was the actual workshop ID already. And in another, it was 2 files, `post.pak` and `contents.pak`, both being necessary for servers [as per the mods workshop page](https://steamcommunity.com/sharedfiles/filedetails/?id=764887546)

Here is the formated relevant code from the egg's [installation script](https://github.com/parkervcp/eggs/blob/master/game_eggs/steamcmd_servers/starbound/egg-starbound.json#L27), right at the end
```bash
## use Worksop content
if [ "${WORKSHOP}" == "1" ]; then
    for dir in /mnt/server/steamapps/workshop/content/211820/*/
    do
     dir=${dir%*/}
     echo Sym-linking mod ${dir##*/} into the mods folder
     ln -r -s /mnt/server/steamapps/workshop/content/211820/${dir##*/}/contents.pak /mnt/server/mods/${dir##*/}.pak
    done
fi
```

As you can see there, it creates a symlink between each subscribed item's contents.pak file into the mods directory

This would normally work, however not every mod has a contents.pak file. A couple of workshop mods do not have just one single `contents.pak` file, sometimes having multiple, or a different name. The filename can be anything, though it is admittedly usually `contents.pak`

I have also seen some of these folders even including a `.DS_Store` file after being freshly subscribed to

### Steps To Reproduce

Create a starbound server
Install a mod like `764887546` which has 2 .pak files, `post.pak` and `contents.pak`, or `862287085` which has a wrongly named .pak file
The code does not copy over these files as it is written to only link `contents.pak` files

### Install logs

https://pteropaste.com/2uigw5/

### PR DESCRIPTION
**Pull request by Wazbat**
Updated script to handle multiple or incorrectly named .pak files

# Description

As described in #1827 the starbound server's install script does not correctly symlink mods if they do not consist of a single `contents.pak` file. I have updated the install script to loop over every single .pak file in the mod's folder, and create a symbolic link for each file, with an incrementing counter on the end in case of multiple .pak files.

I have updated the script and tested it with a new server. I have not modified the exported file, so there are other changes that the panel included.

Resolves #1827 

## Checklist for all submissions

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

### CODE REVIEW

**Comment by parkervcp** [comment]
I can look at this later today.

**Comment by Wazbat** [comment]
Cheers!

Tested it and it seems to work ok. The only thing is I did leave in a `echo $pakfile` line where it creates the symlink. Not sure if that'd be a problem or if you'd want to change that

File: game_eggs/steamcmd_servers/starbound/egg-starbound.json [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
     "author": "parker@parkervcp.com",
     "description": "Starbound takes place in a two-dimensional, procedurally generated universe which the player is able to explore in order to obtain new weapons, armor, and items, and to visit towns and villages inhabited by various intelligent lifeforms.",
     "features": [
         "steam_disk_space"
     ],
-    "images": [
-        "quay.io\/parkervcp\/pterodactyl-images:ubuntu_source"
-    ],
+    "docker_images": {
+        "quay.io\/parkervcp\/pterodactyl-images:ubuntu_source": "quay.io\/parkervcp\/pterodactyl-images:ubuntu_source"
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/starbound/egg-starbound.json:15]: Can you please try if the newer images works for this: `ghcr.io/parkervcp/games:source`

File: game_eggs/steamcmd_servers/starbound/egg-starbound.json [hunk 2]
```diff
@@ -24,7 +24,7 @@
...
             "container": "debian:buster-slim",
```
Comment (gOOvER) [inline game_eggs/steamcmd_servers/starbound/egg-starbound.json:28]: use the Debian Installer image from GHCR

**Review by gOOvER — CHANGES_REQUESTED** [review]

File: game_eggs/steamcmd_servers/starbound/egg-starbound.json (hunk 2 above)
Comment (Wazbat) [reply game_eggs/steamcmd_servers/starbound/egg-starbound.json:28]: > use the Debian Installer image from GHCR

Are you refering to the `ghcr.io/parkervcp/games:source` image that @QuintenQVD0 mentioned? I'm not sure what image I should use as that one doesn't mention debian

Comment (Wazbat) [reply game_eggs/steamcmd_servers/starbound/egg-starbound.json:28]: That, or @parkervcp could try changing it after testing my changes maybe?

Comment (QuintenQVD0) [reply game_eggs/steamcmd_servers/starbound/egg-starbound.json:28]: @Wazbat the install image should the GHCR debian install image. And the game image you should try the 1 I posted and if it works you should use that one

**Comment by Wazbat** [comment]
As requested, I have updated the docker image to `ghcr.io/parkervcp/games:source`. I also removed the extra `echo` from the install script

I tested it and it appears to run ok. Not sure if there's a reduction in memory usage with this image as I didn't note down the old memory usage. Appears to sit around 508MB on a vanilla server without mods

File: game_eggs/steamcmd_servers/starbound/egg-starbound.json (hunk 2 above)
Comment (gOOvER) [reply game_eggs/steamcmd_servers/starbound/egg-starbound.json:28]: > That, or @parkervcp could try changing it after testing my changes maybe?

i pushed a PR to your fork

Comment (Wazbat) [reply game_eggs/steamcmd_servers/starbound/egg-starbound.json:28]: Gotcha, just gonna leave this link here for reference

Wazbat/eggs#1

Comment (Wazbat) [reply game_eggs/steamcmd_servers/starbound/egg-starbound.json]: Ahhh I see what you mean now. I was a bit confused why your code review was on this line. I saw the diff and I thought you were refering to the `script` property, not the `container`

**Comment by Wazbat** [comment]
Merged PR from @gOOvER which updates the installation script image, as well as removing package update and install lines from the top of the install script
```diff
#!/bin/bash
# steamcmd Base Installation Script
- #
- # Server Files: /mnt/server
- # Image to install with is 'ubuntu:18.04'
- apt -y update
- apt -y --no-install-recommends install curl lib32gcc1 ca-certificates

## just in case someone removed the defaults.
if [ "${STEAM_USER}" == "" ]; then
(...)
```

**Comment by gOOvER** [comment]
> Merged PR [Wazbat#1](https://github.com/Wazbat/eggs/pull/1) from @gOOvER which updates the installation script docker image to `ghcr.io\/parkervcp\/installers:debian`, as well as removing package update and install lines from the top of the install script
> 
> ```diff
> #!/bin/bash
> # steamcmd Base Installation Script
> - #
> - # Server Files: /mnt/server
> - # Image to install with is 'ubuntu:18.04'
> - apt -y update
> - apt -y --no-install-recommends install curl lib32gcc1 ca-certificates
> 
> ## just in case someone removed the defaults.
> if [ "${STEAM_USER}" == "" ]; then
> (...)
> ```

you dont need this, because both are in installimage allready.

**Comment by Wazbat** [comment]
> you dont need this, because both are in installimage allready.
> 
> lib32gcc1 also changed to lib32gcc-s1

Yeah that makes complete sense. Just commenting it here for visibility.

Comparing the bash scripts when they're just a single line with multiple escaped characters is a pain haha

**Comment by gOOvER** [comment]
what i missed is to update the installscript to newest

https://github.com/parkervcp/eggs/blob/master/scripts/steamcmd_installer.sh

**Comment by Wazbat** [comment]
What's the status on this? Don't want to let it get stale

**Comment by parkervcp** [comment]
I hadn't had a chance to parse the scripts until now. It looks fine. I see what was changed at least.

Easiest way to compare the scripts is to pull the string and do a replace on `\/` with `/` and `\r\n` with a new line which expands them and makes them readable.

**Comment by Wazbat** [comment]
> Easiest way to compare the scripts is to pull the string and do a replace on `\/` with `/` and `\r\n` with a new line which expands them and makes them readable.

Yep! That seems to work, though you also have to un-escape the double quotes too

I wonder if you could do a simple github action step that would post in a comment a diff view of the scripts if that line was modified. Curious if there'd be any need for it. Could be a fun project

**Review by parkervcp — APPROVED** [review]

**Comment by Wazbat** [comment]
> I might be able to swing something on my end.

Nice, yeah that'd be cool to see.

Hardest part would be checking the diff to see if that line in an egg file was updated, but it should be doable. Another thing is keeping the comment up to date, should the author add another commit to the PR that modifies the script again, though I believe there is an activity type that would allow you to keep checking that

**Pull request merged**